Misc/NEWS.d/next/Security/2023-01-24-16-12-00.gh-issue-101283.9tqu39.rst

```diff
@@ -0,0 +1,6 @@
+Windows version of :meth:`subprocess.Popen.__init__` with
+``shell=True`` made its shell search algorithm protected from dropping
+a program named ``cmd.exe`` into a current directory. Now the current
+directory is used as the last resort if system environment variables
+are totally stripped and Windows is installed not into ``C\Windows``.
```

Comment:
How should we document the change in `docs.python.org`? I suspect it cannot go under "Changed in Changed in version 3.x" because it will span through ≥3.7.17, ≥3.8.17, ≥3.9.17, ≥3.10.9, and ≥3.11.2.

Reply:
Backporting this kind of change requires manual effort, unfortunately. List all the changed versions in `main`, then as we backport we'll drop any versions that are newer than the target branch.

Reply:
How should I mention multiple versions?

- `.. versionchanged:: 3.7.17, 3.8.17, 3.9.17, 3.10.9, and 3.11.2`break rendering:

   > ![100](https://user-images.githubusercontent.com/4881073/215251913-f627d4cd-22a0-44d2-b555-cb9b4e73552d.png)

- comma-less `.. versionchanged:: 3.7.17 3.8.17 3.9.17 3.10.9 3.11.2` doesn't work too:

   > ![Безымянный](https://user-images.githubusercontent.com/4881073/215251914-ab391c9a-176b-4cea-bbc4-0a29df74f872.png)

For reference, that's a random single-version entry:

> ![image](https://user-images.githubusercontent.com/4881073/215251512-e05b9d97-d57a-4655-af96-ddd4234d80d7.png)

Reply:
One-per-line versions don't help either:

```
   .. versionchanged:: 3.7.17
   .. versionchanged:: 3.8.17
   .. versionchanged:: 3.9.17
   .. versionchanged:: 3.10.9
   .. versionchanged:: 3.11.2

      Changed Windows shell search [...]
```

> ![](https://user-images.githubusercontent.com/4881073/215252600-f5adf987-5d0a-474c-9b9e-a5c8dc0de3a7.png)

Reply:
Just mention the current main branch version (3.12), it is normal to manually edit these documentation tags as appropriate in each of the bugfix backports.

Reply:
@gpshead Thank you for the advice; applied.